app/views/shared/vote/_active_button_options.html.erb

```diff
@@ -0,0 +1,9 @@
+<% if !user_signed_in? %>
+  <%= render partial: "shared/vote/static_icons", locals: { parent: parent } %>
+<% elsif !parent.users_vote(current_user).present? %>
+  <%= render partial: "shared/vote/up_or_down", locals: { parent: parent } %>
+<% elsif parent.users_vote(current_user).value == 1 %>
+  <%= render partial: "shared/vote/up_to_down", locals: { parent: parent } %>
+<% else %>
+  <%= render partial: "shared/vote/down_to_up", locals: { parent: parent } %>
+<% end %>
```

Comment:
these nested partials and conditionals are a little hard to follow. could the icon color be driven by a `disabled` state?  could `link_to_if` be used based on 1) whether the user is authenticated and 2) what their current vote value is for a given thing?